components/common/src/templating/mod.rs

```diff
@@ -91,6 +95,35 @@ impl TemplateRenderer {
             .render(template, &raw)
             .map_err(|e| Error::TemplateRenderError(format!("{}", e)))
     }
+
+    // This method is only implemented so we can intercept the call to Handlebars and display
+    // a deprecation message to users. More information here https://github.com/habitat-sh/habitat/issues/6323.
+    // When Handlebars is upgraded this can be safely removed.
+    pub fn register_template_file(&mut self,
```

Comment:
In general, I'm a fan of the simpler approach like you employed with the `path` argument (`&Path` rather than `P: AsRef<Path>`). But in this case, since we're essentially overriding [a method of the `Handlebars` struct we're wrapping](https://docs.rs/handlebars/0.29.1/handlebars/struct.Handlebars.html#method.register_template_file), I think it makes sense to keep the signature the same. Especially since the change as is breaks existing tests which expect the other signature.